spec.bs

```diff
@@ -2942,17 +3003,29 @@ of the following global objects:
     1. If |result| is a [=ECMAScript/normal completion=]:
       1. Let |generatedBidIDL| be the result of [=converted to an IDL value|converting=]
         |result|'s \[[Value]] to a {{GenerateBidOutput}}.
-      1. If no exception was [=exception/thrown=] in the previous step, then set |generatedBid| to
-        the result of [=converting GenerateBidOutput to generated bid=] with |generatedBidIDL|,
-        |ig|, |expectedCurrency|, |isComponentAuction|, and |global|'s
-        [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=].
-      1. Otherwise, set |generatedBid| to failure.
+      1. If no exception was [=exception/thrown=] in the previous step:
+        1. Let |bidOutput| to the result of [=converting GenerateBidOutput to generated bid=] with
+          |generatedBidIDL|, |ig|, |expectedCurrency|, |isComponentAuction|, and |global|'s
+          [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=].
+        1. If |bidOutput| is not failure, then set |generatedBid| to |bidOutput|.
+    1. Let |debugLossReportUrl| be |global|'s
+      [=InterestGroupBiddingAndScoringScriptRunnerGlobalScope/debug loss report url=] if it's not
+      failure, null otherwise.
     1. If |generatedBid| is a [=generated bid=] and |generatedBid|'s [=generated bid/bid=]'s
-      [=bid with currency/value=] &le; 0, set |generatedBid| to failure.
-    1. If |generatedBid| is null, set it to failure.
-    1. If |generatedBid| is not failure:
-      1. Set |generatedBid|'s [=generated bid/bid duration=] to |duration|.
-      1. Set |generatedBid|'s [=generated bid/interest group=] to |ig|.
+      [=bid with currency/value=] &le; 0, then set |generatedBid| to null.
+    1. If |generatedBid| is a [=generated bid=]:
+      1. Let |debugWinReportUrl| be |global|'s
+        [=InterestGroupBiddingAndScoringScriptRunnerGlobalScope/debug win report url=] if it's not
+        failure, null otherwise.
+      1. Set |generatedBid|'s [=generated bid/bid duration=] to |duration|,

```

Comment:
@domfarolino Is it OK to set multiple fields like this? Wanted to avoid having too many "Set ..." steps.

Reply:
Yeah this is fine, and you can do that here too: https://github.com/WICG/turtledove/pull/977/files#diff-6f5a1d8263b0b0c42e2716ba5750e3652e359532647ac934c1c70086ae3ceddaR1700-R1706